src/state.rs

```diff
@@ -456,6 +517,43 @@ impl DemandProxyState {
             .cloned()
     }
 
+    pub fn set_cached_resolve_dns_for_hostname(
+        &mut self,
+        hostname: String,
+        cached_by_hostname: SingleFlight,
+    ) {
+        let mut binding = self.state.write().unwrap();
+        let sf_map_element = binding
+            .resolved_dns
+            .sf_by_hostname
+            .entry(hostname)
+            .or_insert(cached_by_hostname);
+        // increment 1 for every requests in concurrency
+        sf_map_element.calling_index += 1;
```

Comment:
sf_map_element is cloned, this modify does not affect the original element

Reply:
Hi @hzxuzhonghu, this modification does affect the original element because `or_insert` will return a mutating reference of this element in Hashmap.

Reply:
However, I also should add `*` for `sf_map_element`, so it should be like this:
`(*sf_map_element).calling_index += 1;`. Thanks!